stage2/tasks/fun-chat/README.md

```diff
@@ -0,0 +1,156 @@
+- (+5) Sending a message to a user is possible both by clicking the send button with the mouse or by pressing the "Enter" key without the need to focus on the send button.
+- (+10) The user can delete their own previously sent messages.
+- (+10) The user can edit the text of their own previously sent messages.
+
+#### 7. Message Content (+25)
+
+- (+15) Messages include the time of sending, sender's username, message delivery status, message text, and indication of whether the message has been edited.
+- (+5) The message "delivered"/"read" status is visible only to the sender of the message.
+- (+5) A user cannot send a message without any content (without text).
+
+#### 8. Message Delivery and Read Status (+20)
+
+- (+20) The status changes to "delivered" when the message recipient logs into the application or if the message is sent to the user who is online.
+- (+20) The status changes to "read" when the message recipient opens a dialogue with unread messages and performs any of the following actions: scrolls in the message history area, sends a new message, or clicks inside the message history area.
+
+#### 9. Interface and Visual Design (+15)
+
+- (+5) The browser tab must display the application icon.
+- (+5) The interface elements with which the user will interact must be responsive and the cursor must change when they are hovered over.
+- (+5) Responsive layout must be implemented for resolutions ranging from 1440 px to 380 px, inclusive.
+
+#### 10. Server Connection (+20)
+
+- (+10) If a sudden disconnection from the server occurs, a message must be displayed to the user, and an attempt to restore the connection must be made.
+- (+10) Upon reconnecting to the server, the application must perform the current user reauthorization without requiring user intervention.
+
+## Technical Requirements (+160)
+
+- (+20) The application is divided into logical modules/layers, such as, for example, API interaction, user interface rendering, application state management, etc.
+- (+20) All HTML content is generated using JavaScript. `Body` must be empty. Either `head` or `body` can contain the only `<script> tag` (`body` containing only the `<script> tag` is considered to be empty).
+- (+20) The application is a Single Page Application (SPA) with implemented routing.
+- (+20) The application is implemented using TypeScript.
+- (+20) Input/output parameters of all methods are explicitly typed, and the `any` type is not used.
+- (+20) ESLint with the Airbnb style guide is used, with the `noInlineConfig: true` rule enabled in the configuration.
+- (+10) Webpack or another module bundler is used.
+- (+10) Code is organized into small functions with clear names and purposes, with each function not exceeding 40 lines.
+- (+10) There is no code duplication.
+- (+10) The code does not contain magic numbers or strings.
+
```

Comment:
We can add items about setting up scripts for Prettier and ESLint. I will describe this in another task, and then we can transfer it to your task

Reply:
The use of ESLint is specified [above](https://github.com/rolling-scopes-school/tasks/pull/1502/files#diff-0fdef4afb7b7cb4ad8314f116a1cf2458a069d03a56369a0032d849e9b0a0c78R107), the use of Prettier we will add

Reply:
I meant exactly setting up scripts for Prettier and ESLint in package.json. Students often just install and use only the VSCode plugins